spec/rails_helper.rb

```diff
@@ -21,6 +21,8 @@
 RSpec.configure do |config|
   config.include FactoryBot::Syntax::Methods
 
+  config.include Warden::Test::Helpers # fake devise user for feature tests
```

Comment:
Should we qualify this to  `type: :feature`?

I'm also wondering why you chose not to use the `Devise::TestHelpers` instead?  These operate at a higher level than Warden.

Reply:
i didn't know about them, but now I do - thanks!